### PR TITLE
feat: add docker dev support [#133]

### LINKED ISSUE #133: [Feat] Add support for Docker development
There is no guide for developing on Mac, let alone Docker. Adding support for this will allow developers from most platforms to work in a consistent environment

### PR DESCRIPTION
**Pull request by nickatnight**
This PR allows developers to work in an isolated container environment. Follow the [Docker Guide](https://github.com/dogecoin/dogecoin.com/pull/134/files#diff-10d848450532cb6ae404872f68604593da6ea0d670fc9b482a2675b9cacefa38) to get started.

The container is built from a Node base image and only needs to install a single dependency, `live-server`. Running the container will boot up a local web server which is available from the host machine on `localhost:5000` or `127.0.0.1:5000`

Something to think about for the future is to modify the folder structure and move all site files to a dedicated`src` folder. Then build some minified dist bundle so files will decrease in size

### CODE REVIEW

**Comment by patricklodder** [comment]
Thanks. Since this is a GitHub-pages repo, I think it's okay to keep the website source as the root, and instead put all support files in a subfolder (with a leading underscore so that it doesn't get published.)

Can we move the Dockerfile, and npm descriptors to `_contrib/docker`? Then you simply `docker build _contrib/docker` instead of the local dir, and I think you could omit the `ADD` as well? Also reduces image size.

**Comment by nickatnight** [comment]
@patricklodder thank you for your response. Made some changes:

1. removed npm descriptor files since we need only that single dependency, and just installed `live-server` command globally.
2. moved `Dockerfile` to appropriate sub folder
3. updated docker contribution file

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,6 @@ site/_site/
 coverage
 .ruby-version
 .sass-cache
+
+# Distribution / packaging
+node_modules
```
Comment (patricklodder) [inline .gitignore]: I think we can remove these.

**Review by patricklodder — CHANGES_REQUESTED** [review]
One nit: there's no need to exclude node_modules anymore with the global install of live-server.

Everything else: works perfectly - thank you! 👍

**Comment by nickatnight** [comment]
@patricklodder the deed is done chief 🍻

**Review by patricklodder — APPROVED** [review]

**Pull request merged**